Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc

```diff
@@ -928,10 +1032,69 @@ double HcalIsoTrkAnalyzer::rhoh(const edm::Handle<CaloTowerCollection>& tower) {
   else evt_smdq = (sumPFNallSMDQH2[sumPFNallSMDQH2.size()/2]+sumPFNallSMDQH2[(sumPFNallSMDQH2.size()-2)/2])/2.;
   double rhoh = evt_smdq/(etadist_*phidist_);
 #ifdef EDM_ML_DEBUG
-  edm::LogInfo("HcalIsoTrack") << "Rho " << evt_smdq << ":" << rhoh;
+  edm::LogVerbatim("HcalIsoTrack") << "Rho " << evt_smdq << ":" << rhoh;
 #endif
   return rhoh;
 }
 
+void HcalIsoTrkAnalyzer::storeEnergy(int indx, const HcalRespCorrs* respCorrs, 
+				     const std::vector<DetId>& ids,
+				     std::vector<double>& edet,
+				     double & eHcal, 
+				     std::vector<unsigned int> *detIds, 
+				     std::vector<double> *hitEnergies) {
+  double ehcal(0);
+  if (unCorrect_) {
+    for (unsigned int k=0; k<ids.size(); ++k) {
+      double corr = (respCorrs->getValues(ids[k]))->getValue();
+      if (corr != 0) edet[k] /= corr;
+      ehcal += edet[k];
+    }
+  } else {
+    for (auto en : edet) ehcal += en;
+  }
+  if (std::abs(ehcal-eHcal) > 0.001) 
+    edm::LogWarning("HcalIsoTrack") << "Check inconsistent energies: " << indx
+				    << " " << eHcal << ":" << ehcal
+				    << " from " << ids.size() << " cells";
+  eHcal = hcalScale_*ehcal;
+
+  if (collapseDepth_) {
+    std::map<HcalDetId,double> hitMap;
+    for (unsigned int k=0; k<ids.size(); ++k) {
+      HcalDetId id = hdc_->mergedDepthDetId(HcalDetId(ids[k]));
+      auto itr = hitMap.find(id);
+      if (itr == hitMap.end()) {
+	hitMap[id] = edet[k];
+      } else {
+	(itr->second) += edet[k];
+      }
+    }
+    for (auto hit : hitMap) {
+      detIds->emplace_back(hit.first.rawId());
```

Comment:
add reserve calls for detIds and hitEnergies

Reply:
done